cfme/tests/automate/test_common_methods.py

```diff
@@ -250,3 +253,73 @@ def test_service_retirement_from_automate_method(request, generic_catalog_item,
         description=f"Service Retire for: {generic_catalog_item.name}")
     wait_for(lambda: retire_request.request_state == "finished", fail_func=retire_request.reload,
              timeout=180, delay=10)
+
+
+@pytest.fixture
+def set_roottenant_quota(request, appliance):
+    field, value = request.param
+    roottenant = appliance.collections.tenants.get_root_tenant()
+    roottenant.set_quota(**{'{}_cb'.format(field): True, field: value})
+    yield
+    roottenant.set_quota(**{'{}_cb'.format(field): False})
+
+
+@pytest.mark.tier(3)
+@pytest.mark.usefixtures("setup_provider")
+@pytest.mark.meta(automates=[1334318])
+@pytest.mark.provider([VMwareProvider], override=True)
+@pytest.mark.parametrize(
+    ['set_roottenant_quota'],
+    [
+        [('memory', '1000')],
+    ],
+    indirect=['set_roottenant_quota'],
+    ids=['memory']
+)
+def test_automate_quota_units(request, set_roottenant_quota, provisioning, provider, appliance):
+    """
+    Bugzilla:
+        1334318
+
+    Polarion:
+        assignee: ghubale
+        casecomponent: Automate
+        caseimportance: low
+        initialEstimate: 1/4h
+        tags: automate
+    """
+    vm_name = random_vm_name(context='quota')
+
+    prov_data = {
+        "catalog": {'vm_name': vm_name},
+        "environment": {'automatic_placement': True},
+        "network": {'vlan': partial_match(provisioning['vlan'])},
+        'hardware': {'memory': '1024'},
+    }
+
+    def _finalize():
+        collection = appliance.provider_based_collection(provider)
+        vm_obj = collection.instantiate(vm_name, provider, provisioning["template"])
+        try:
+            vm_obj.cleanup_on_provider()
+        except Exception:
+            logger.warning('Failed deleting VM from provider: %s', vm_name)
+
+    request.addfinalizer(_finalize)
```

Comment:
**Suggestion:**
Better to use single line `request.addfinalizer()` for inline functions. You can use decorator like this:

```
@requsest.addfinalizer
def _finalize():
    collection = appliance.provider_based_collection(provider)
    vm_obj = collection.instantiate(vm_name, provider, provisioning["template"])
    try:
        vm_obj.cleanup_on_provider()
    except Exception:
        logger.warning('Failed deleting VM from provider: %s', vm_name)
```